### DISCUSSION TITLE
[jasmine]: adding a custom matcher means jumping through some hoops

### DISCUSSION THREAD

**Discussion by ajvincent** [DefinitelyTyped/DefinitelyTyped#60935] [Issues with a @types package]
```typescript

type AddedCustomMatchers<T extends jasmine.CustomMatcherFactories> = {
  [P in keyof T]: (expected: Parameters<ReturnType<T[P]>["compare"]>[1]) => void;
}

type FieldMatcherMethods = AddedCustomMatchers<FieldMatchers>;

// ...

(expect(repeatForward) as jasmine.Matchers<any> & FieldMatcherMethods).toBeMethodDefinition({
  type: AST_NODE_TYPES.MethodDefinition,
  static: false,
  isPrivate: false,
  kind: "method"
});
```

1. `jasmine.Matchers` didn't provide any parameter type for extensions, so TypeScript explicitly requires the intersection type.
2. That `AddedCustomMatchers` type is _ugly_.  It boils down to `(expected: any) => void`, sure, but I was really hoping to extract the rest of the `CustomMatcher` interface in a way that was somewhat nicer.  (I'm very surprised that my implementation only required the non-parameterized compare method of the `CustomMatcher`, but whatever.)
3. `// eslint-disable-next-line @typescript-eslint/no-explicit-any`

If I'm missing something obvious, please tell me...

[Full code sample](https://github.com/ajvincent/cross-stitch/blob/77e7f36ae6ad3c33d347f6fadb9b8df5dcb4283f/_02_TypeScript_TypeToClass/spec/_b_generated/NotImplementedClasses.mts)

[a comment by an automated account is not shown]

**Comment by chivesrs** [comment]
I thought for custom matchers you were supposed to reopen the jasmine namespace.

https://stackoverflow.com/questions/42956195/create-custom-jasmine-matcher-using-typescript

This looks about what I have in my code base

**Reply by ajvincent** [reply]
That would be the "missing something obvious" part, I guess.  I thought namespaces were deprecated in TypeScript.

Could I ask for some code documentation comments to be added to the types file for this particular problem?  I'm sorry to fill this place with spam, but I'm clearly not the first person to ask this.

**Comment by chivesrs** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jasmine/jasmine-tests.ts#L1850

The tests cover this already. You can open a PR that adds a comment in the main typings to say to look at the tests.
